### ISSUE TITLE
Disable SMTP TLS verify

### ISSUE THREAD

**Issue by ajgon** [nitnelave/lldap#412]
I'm using custom SMTP relay in my k8s cluster, with self-signed certificates. Unfortunatelly, I can't make LLDAP o speak with it, I'm getting this error:

```
2023-01-07T14:37:29.958857006+00:00 WARN     │  ┝━ 🚧 [warn]: Sending fatal alert BadCertificate | log.target: "rustls::conn" | log.module_path: "rustls::conn
" | log.file: "/__w/lldap/lldap/${GITHUB_WORKSPACE}/.cargo/registry/src/github.com-1ecc6299db9ec823/rustls-0.20.6/src/conn.rs" | log.line: 1285
2023-01-07T14:37:29.959013837+00:00 WARN     │  ┕━ 🚧 [warn]: Error sending email: lettre::transport::smtp::Error {
    kind: Connection,
    source: lettre::transport::smtp::Error {
        kind: Connection,
        source: Custom {
            kind: InvalidData,
            error: InvalidCertificateData(
                "invalid peer certificate: UnknownIssuer",
            ),
        },
    },
}
```

Is it possible to disable cert verification (or TLS/STARTTLS altogether) ?

**Comment by nitnelave** [comment]
It should be possible to provide the certificate in the config, and configure lettre (the email lib) to accept that as an additional root certificate in the TLS parameters.

It's not trivial, but it's possible.

In the meantime, why not get a properly signed certificate with letsencrypt?

**Comment by ajgon** [comment]
The problem is, I'm using relay inside k8s cluster - which means, I'm using k8s domains like `relay.mail.svc.cluster.local` etc. - which can't be (as far as I'm aware of) registered in LE. And fiddling with that system is cumbersome, and does more harm than good in my opinion.

Passing custom certs is the "proper" way of doing this, but again it's a PITA in k8s (you need to create certs in one pod, create secrets, migrate them through namespaces etc.), and (again in my opinion) not worth the effort.

The simplest and widely accepted solution in custom networks as this, is just disable SSL verification. You traffic is still encrypted, and you just trade convenience for possibility of MITM attack. Also if somebody can MITM inside your cluster, you have way bigger problem than disabled verification :)

**Comment by nitnelave** [comment]
Yeah, I looked at the docs again, and it's possible to disable certificate
checks.
How do you feel about trying to do a PR? It wouldn't require a lot of Rust.

**Comment by ajgon** [comment]
Funny thing is, I've written some code in Rust long time ago... but don't remember a thing :D Maybe it's worth to do a little refresh... I'll try, and see where it gets me :)

**Comment by nitnelave** [comment]
I'm sure you'll do fine, it shouldn't be too complicated. A few tips:
 - the email sending code is in server/src/infra/mail.rs
 - you want to build TLS parameters to pass to the SMTP relay. In the TLS
builder you have a `dangerous_` method to disable the certificate checks.
 - you'll want to add a config variable. That's in server/src/infra/
configuration.rs and cli.rs (for the command line override flag). Also add
it to the docker config example at the top level. It should probably be a
positive flag (enable_certificate_check?) defaulting to true
 - there are no tests for that part of the code, so test by hand.

That should be enough to get you started, don't hesitate to ask here or in
discord if you need help!

**Comment by ajgon** [comment]
Hmm... looks like, envs are also broken:
```bash
$ LLDAP_SMTP_OPTIONS__ENCRYPTION=STARTTLS tmp/release/x86_64/lldap send_test_email -v --to test@example.net | grep smtp_options -A10
    smtp_options: MailOptions {
        enable_password_reset: false,
        from: None,
        reply_to: None,
        server: "localhost",
        port: 587,
        user: "admin",
        password: ***SECRET***,
        smtp_encryption: TLS,
        tls_required: None,
    },
```

but config works :)

```bash
$ cat test.toml
verbose=true

[smtp_options]
smtp_encryption = "STARTTLS"

$ tmp/release/x86_64/lldap send_test_email -c test.toml --to test@example.net | grep smtp_options -A10
    smtp_options: MailOptions {
        enable_password_reset: false,
        from: None,
        reply_to: None,
        server: "localhost",
        port: 587,
        user: "admin",
        password: ***SECRET***,
        smtp_encryption: STARTTLS,
        tls_required: None,
    },
```

**Comment by nitnelave** [comment]
Solved with #423
